### DISCUSSION TITLE
[Bug] backend storage(elasticsearch) cannot change number_for_shards

### DISCUSSION THREAD

**Discussion by ox-warrior** [apache/skywalking#9222] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

elasticsearch number_of_shards not work

### What you expected to happen

when i change 「SW_STORAGE_ES_SUPER_DATASET_INDEX_SHARDS_FACTOR」to 6 &「SW_STORAGE_ES_INDEX_SHARDS_NUMBER」to 1 
the new index like 「sw_segment」's number_of_shards need to be 6 ,but it's still was "5"

### How to reproduce

skywalking oap server version: 8.9.1
elasticsearch version: elasticsearch-oss:7.10.2-arm64

### Anything else

when i delete the template:sw_segment, and create index:sw_segment-hello it's still use number_of_shards=5

i change another elasticsearch cluster(without skywalking template ), the new index also was 5 number_of_shards

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Yes, this is expected behavior. There is no point to use SkyWalking OAP to change the existing template settings.

**Reply by ox-warrior** [reply]
but when i delete the template it's still can create index with number_of_shards=5 
is OAP do somethings else to elasticsearch cluster, only the super data set index has this problem

**Reply by wu-sheng** [reply]
It creates templates if they don't exist, but it would not change it. 

> is OAP do somethings else to elasticsearch cluster, only the super data set index has this problem

No, it isn't. All indices follow this principle.
